src/NuGet.Core/NuGet.Commands/PackCommandRunner.cs

```diff
@@ -244,26 +245,38 @@ private static void AddDependencyGroups(IList<LibraryDependency> dependencies, N
                     continue;
                 }
 
-                List<string> includes = new List<string>();
-                if (effectiveInclude == LibraryIncludeFlags.All)
+                if (isTargetFrameworks && dependency.LibraryRange.TypeConstraint == LibraryDependencyTarget.Project)
                 {
-                    includes.Add(LibraryIncludeFlags.All.ToString());
+                    continue;
                 }
-                else if ((effectiveInclude & LibraryIncludeFlags.ContentFiles) == LibraryIncludeFlags.ContentFiles)
+                else if (dependency.LibraryRange.TypeConstraint == LibraryDependencyTarget.Reference)
                 {
-                    includes.Add(LibraryIncludeFlags.ContentFiles.ToString());
+                    builder.FrameworkReferences.Add(new FrameworkAssemblyReference(dependency.Name, new NuGetFramework[] { framework }));
```

Comment:
A framework reference could be listed under multiple frameworks, is the list combined later on?